fuse/plugins/micro_physics/yields.py

```diff
@@ -174,75 +167,57 @@ class BetaYields(NestYields):
     provides = "quanta"
     data_kind = "interactions_in_roi"
 
-    # Config options
-
-    use_recombination_fluctuation = straxen.URLConfig(
-        default=True,
-        type=bool,
-        help="Turn on or off the recombination fluctuation for beta interactions.",
-    )
-
-    recombination_fluctuation_std_factor = straxen.URLConfig(
-        default=3,
-        type=(int, float),
-        help="A factor that is defined to guess the recombination fluctuation",
-    )
-
     beta_quanta_spline = straxen.URLConfig(
         default=None,
         help="Path to function that gives n_ph and n_e for a given energy, \
         calculated from beta spectrum. The function should be a pickle file.",
     )
 
-    def get_quanta(self, interactions_in_roi):
-
-        # for the non beta interactions we use the nestpy yields
-        photons, electrons, excitons = self.quanta_from_NEST(
-            interactions_in_roi["ed"],
-            interactions_in_roi["nestid"],
-            interactions_in_roi["e_field"],
-            interactions_in_roi["A"],
-            interactions_in_roi["Z"],
-            interactions_in_roi["create_S2"],
-            density=interactions_in_roi["xe_density"],
-        )
+    beta_yield_threshold = straxen.URLConfig(
+        default=10,
+        help="Threshold in keV above which we apply the beta quanta spline.",
+    )
 
-        mask_beta = interactions_in_roi["nestid"] == 8
+    def setup(self):
 
-        # now for the beta interactions we use the beta yields
-        photons_beta, electrons_beta = self.quanta_from_spline(
-            interactions_in_roi["ed"][mask_beta],
-        )
+        if self.beta_quanta_spline is None:
+            raise ValueError("beta_quanta_spline must be set in the context config")
 
-        photons[mask_beta] = photons_beta
-        electrons[mask_beta] = electrons_beta
+        super().setup()
```

Comment:
Do we need this explicit check if the spline is set? I guess strax will give us an error message anyway or?